export.py

```diff
@@ -2844,41 +2852,58 @@ def export_display(ri, rpass, scene):
+            if rm_rl.export_multilayer and rpass.external_render:
+                channels = []
+                for aov in rm_rl.custom_aovs:
+                    channels.append(
+                        aov.channel_name) if aov.channel_type != "Ci,a" else channels.append("Ci,a")
                 out_type, ext = ('openexr', 'exr')
-                # removes 'z' and 'zback' channels as DeepEXR will automatically add them
+                # removes 'z' and 'zback' channels as DeepEXR will
+                # automatically add them
                 if rm_rl.use_deep:
-                    channels = [x for x in channels if x not in ['z', 'zback']]
+                    channels = [x for x in channels if x not in [
+                        'z_depth', 'z_back']]
                     out_type, ext = ('deepexr', 'exr')
                 params = {"string storage": rm_rl.exr_storage}
                 if rm_rl.exr_format_options != 'default':
                     params["string type"] = rm_rl.exr_format_options
                 if rm_rl.exr_compression != 'default':
                     params["string compression"] = rm_rl.exr_compression
                 ri.Display('+' + image_base + '.%s' % layer_name +
-                               '.multilayer.' + ext, out_type, ','.join(channels), params)
+                           '.multilayer.' + ext, out_type, ','.join(channels), params)
 
             else:
                 for aov in rm_rl.custom_aovs:
-                    if layer == scene.render.layers[0] and aov == 'rgba':
-                        #we already output this skip
+                    aov_name = aov.name.replace(' ', '')
+                    if aov.channel_type == "Ci,a":
+                        aov.channel_name = "rgba"
+                    if layer == scene.render.layers[0] and aov == 'Ci,a':
+                        # we already output this skip
                         continue
                     params = {}
                     if not rpass.external_render:
-                        params["int asrgba"] = 1
+                        params = {"int asrgba": 1}
                     if aov.denoise_aov:
                         ri.Display('+' + image_base + '.%s.%s.denoiseable.' %
                                    (layer_name, aov_name) + ext, display_driver, aov.channel_name)
                     else:
-                        dspy_name = image_base + '.%s.%s.' % (layer_name, aov_name) + ext
-                        ri.Display('+' + dspy_name, display_driver, aov.channel_name, params)
+                        dspy_name = image_base + \
+                            '.%s.%s.' % (layer_name, aov_name) + ext
+                        ri.Display('+' + dspy_name, display_driver,
+                                   aov.channel_name, params)
                         rpass.output_files.append(dspy_name)
 
-    if rm.do_denoise and not rpass.is_interactive:
+    if rm.do_denoise or rm.external_denoise and not rpass.is_interactive:
```

Comment:
True.  Incidentally the same thing happens if denoise is checked in the internal render options and an external render is done.  I'll add a check that will output the denoise channels only if the corresponding denoise option is checked for internal or external rendering.